### PR TITLE
improve Python3 support

### PR DESCRIPTION
**Pull request by benoit-pierre**
Note:
* this depend on #28 
* I added `six` as a dependency
* I reworked some changes in `xobject` to keep the API compatible with previous versions
* again, the big-endian tests should be regenerated

### CODE REVIEW

**Comment by benoit-pierre** [comment]
Fixed an issue reported by `code-quality/landscape`.

**Comment by benoit-pierre** [comment]
Updated `six` requirements to `>=1.10.0` after getting errors when testing on Ubuntu 14.04 (Trusty).

**Comment by benoit-pierre** [comment]
Note: if Python 2.6 support is really needed, then one way to fix it for this PR and #28 is to:
* [require `unittest2` for Python `<2.7`](https://github.com/benoit-pierre/python-xlib/commit/cfc20450213f9e8232ef4f5b2eb691278099544e): so we get support for `unittest.SkipTest`
* and [require `importlib` for Python `<2.7`](https://github.com/benoit-pierre/python-xlib/commit/19f837171b6cb6da2c773d6f44918e7041a172be): for the changes in `Xlib/support/connect.py`

File: Xlib/protocol/display.py [hunk 1]
```diff
@@ -25,6 +25,9 @@
 import errno
 import socket
 
+# Python 2/3 compatibility.
+from six import PY3, byte2int, indexbytes
```
Comment (vasily-v-ryabov) [inline Xlib/protocol/display.py]: (Minor) PEP8-complaint import style is
```python
from six import PY3
from six import byte2int
from six import indexbytes
```
`rq.py` has the same issue.

**Comment by vasily-v-ryabov** [comment]
First, thanks a lot for the great job done! I like all the changes except very minor things.

Some examples miss `from __future__ import print_function` statement that will not produce `SyntaxError` but it may change the behavior of prints like that:
```python
>>> print(1, 2)
(1, 2)
>>> from __future__ import print_function
>>> print(1, 2)
1 2
```

File: Xlib/protocol/display.py (hunk 1 above)
Comment (benoit-pierre) [inline Xlib/protocol/display.py]: I can change those, but this is something that is explicitly mentioned as allowed in PEP8:

> It's okay to say this though:
> 
> `from subprocess import Popen, PIPE`

**Comment by benoit-pierre** [comment]
> First, thanks a lot for the great job done! I like all the changes except very minor things.
>
> Some examples miss from __future__ import print_function statement that will not produce SyntaxError but it may change the behavior of prints like that:

OK, amended.

File: Xlib/protocol/display.py (hunk 1 above)
Comment (vasily-v-ryabov) [inline Xlib/protocol/display.py]: Sorry, maybe this is not a PEP8 recommendation but Landscape.io or QuantifiedCode warning (I can't find it quickly right now).

Comment (benoit-pierre) [inline Xlib/protocol/display.py]: It's [not Landscape.io](https://landscape.io/github/benoit-pierre/python-xlib/11/messages?classes=syntax&classes=error&classes=smell&classes=style&classes=docs).

**Pull request merged**